lib/rubocop/cop/rails/where_missing.rb

```diff
@@ -0,0 +1,108 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Rails
+      # Check for the use `left_joins` and `where` method for finding missing relationship records.
+      #
+      # This cop is enabled in Rails 6.1 or higher.
+      #
+      # @example
+      #   # bad
+      #   Post.left_joins(:author).where(authors: { id: nil })
+      #
+      #   # good
+      #   Post.where.missing(:author)
+      #
+      class WhereMissing < Base
+        include RangeHelp
+        extend AutoCorrector
+        extend TargetRailsVersion
+
+        MSG = 'Use `where.missing(:%<association>s)` instead of ' \
+              '`%<left_joins_method>s(:%<association>s).where(%<association>ss: { id: nil })`.'
+        RESTRICT_ON_SEND = %i[left_joins left_outer_joins].freeze
+
+        minimum_target_rails_version 6.1
+
+        # @!method where_node_and_argument(node)
+        def_node_search :where_node_and_argument, <<~PATTERN
+          $(send ... :where (hash <(pair $(sym _) (hash (pair (sym :id) (nil))))...> ))
+        PATTERN
+
+        # @!method missing_relationship?(node)
+        def_node_search :missing_relationship?, <<~PATTERN
+          (pair (sym _) (hash (pair (sym :id) (nil))))
+        PATTERN
+
+        def on_send(node)
+          return unless node.first_argument.sym_type?
+
+          where_node_and_argument(node.ancestors.last) do |where_node, where_argument|
```

Comment:
I'm a bit worried about going back the whole tree (e.g. until class itself). Perhaps we can get only the send_type ancestors (is it possible the chain to have no-send nodes? If so, we should express that in tests as well). Perhaps if we match both `where` and `left_joins` we can cover both cases - left_joins being before or after the where, by just inspecting whether the node contains the other part, and avoid going up the whole tree?

Note that with the current implementation, if the file has an offense, but there is another instance of left_joins to the same table, it would mark the other as offence as well, and would wrongly auto-correct it, e.g.
```
def test 
  Post.left_joins(:author).any?

  do_something

  Post.left_joins(:author).where(authors: {id: nil})
end
```
So this can produce false positives across the file, for a single offence would mark all left joins to the same table

Reply:
Indeed it is. Thank you so much.
I added a test and fixed the problem that was causing false positives across the entire file.